crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/util/GenerateCsvUtil.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.laa.crime.crowncourt.util;
+
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.justice.laa.crime.crowncourt.entity.ReactivatedProsecutionCase;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.List;
+import java.util.Set;
+
+@UtilityClass
+@Slf4j
+public class GenerateCsvUtil {
+
+    private static final String FILE_HEADER = "maat id, case URN, hearing id, previous outcome, previous outcome date, date of status change" + System.lineSeparator();
+    private static final String FILE_PERMISSIONS = "rwx------";
+
+    public File generateCsvFile(List<ReactivatedProsecutionCase> reactivatedCaseList, String fileName) throws IOException {
+        File targetFile = createCsvFile(fileName);
+        try (FileWriter fw = new FileWriter(targetFile, true)) {
+            // Write the CSV header
+            fw.append(FILE_HEADER);
+            // Write each ReactivatedProsecutionCase to the CSV file
+            writeCsvRows(reactivatedCaseList, fw);
+
+        } catch (IOException exception) {
+            log.error("Error creating CSV file - {}", exception.getMessage());
+        }
+        return targetFile;
+    }
+
+    private void writeCsvRows(List<ReactivatedProsecutionCase> reactivatedCaseList, FileWriter fw){
+        reactivatedCaseList.stream()
+                .map(reactivatedProsecutionCase -> reactivatedProsecutionCase.getMaatId() + ","
+                        + reactivatedProsecutionCase.getCaseUrn() + ","
```

Comment:
We should be reusing existing libraries for generating CSVs as they will take care of any edge cases with regards to escaping characters when needed etc.

e.g.
`com.opencsv/opencsv CSVWriter.java`

Reply:
yes but we are Ok with this as we have mimicked the DCES report service.

Reply:
ok, it sounds like the DCES service also has this same issue, consistency is more important than the quality of the solution though I think,